blocks/init/src/Blocks/custom/featured-posts/featured-posts.php

```diff
@@ -24,19 +24,48 @@
 <div class="<?php echo esc_attr($blockClass); ?>" data-items-per-line=<?php echo \esc_attr($itemsPerLine); ?>>
 	<?php
 		$postType = $query['postType'];
+		$taxonomy = $query['taxonomy'];
+		$terms = $query['terms'];
 		$posts = $query['posts'];
 
 		$args = [
 			'post_type' => $postType,
 			'posts_per_page' => $showItems,
 		];
 
+		if ($taxonomy) {
+			if ($terms) {
+				$args['tax_query'][0] = [
+					'taxonomy' => $taxonomy,
+					'field' => 'id',
+					'terms' => array_map(
+						function ($item) {
+							return $item['value'];
+						},
+						$terms
+					),
+				];
+			} else {
+				$args['tax_query'][0] = [ // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+					'taxonomy' => $taxonomy,
+					'field' => 'id',
+					'operator' => 'NOT IN'
+				];
+			}
```

Comment:
```suggestion
                      $args['tax_query'][0] = [
				'taxonomy' => $taxonomy,
				'field' => 'id',
			];
							
			if ($terms) {
				$args['tax_query'][0]['terms'] = array_map(
					function ($item) {
						return $item['value'];
					},
					$terms
				);
			} else {
				$args['tax_query'][0]['operator'] = 'NOT IN'; // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
			}
```

I think this should work too. Just removing the duplicates.